source/mod_particles.f90

```diff
@@ -20,55 +18,133 @@ end subroutine part_allocate
 subroutine part_expand
 end subroutine part_expand
 
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Last modified: 2018-10-31
+!  Moved from maincr. Applies the closed orbit correction to the particle coordinates.
+! ================================================================================================ !
+subroutine part_applyClosedOrbit
+
+  use mod_common
+  use mod_commons
+  use mod_commonmn
+
+  implicit none
+
+  integer j
+
+  if(iclo6 == 2) then
+    xv1(1:napx)   = xv1(1:napx)   +  clo6v(1,1:napx)
+    yv1(1:napx)   = yv1(1:napx)   + clop6v(1,1:napx)
+    xv2(1:napx)   = xv2(1:napx)   +  clo6v(2,1:napx)
+    yv2(1:napx)   = yv2(1:napx)   + clop6v(2,1:napx)
+    sigmv(1:napx) = sigmv(1:napx) +  clo6v(3,1:napx)
+    dpsv(1:napx)  = dpsv(1:napx)  + clop6v(3,1:napx)
+  else if(idfor == 0) then
+    xv1(1:napx)   = xv1(1:napx)   +   clov(1,1:napx) * real(idz(1),fPrec)
+    yv1(1:napx)   = yv1(1:napx)   +  clopv(1,1:napx) * real(idz(1),fPrec)
+    xv2(1:napx)   = xv2(1:napx)   +   clov(2,1:napx) * real(idz(2),fPrec)
+    yv2(1:napx)   = yv2(1:napx)   +  clopv(2,1:napx) * real(idz(2),fPrec)
+  end if
```

Comment:
Not sure. These were copy/pasted from the old code. I only changed the 1:napx indexing.

Reply:
There is a section in maincr that does this:
```
              if(iclo6r == 0) then
                clo6(1)  = clo(1)
                clop6(1) = clop(1)
                clo6(2)  = clo(2)
                clop6(2) = clop(2)
                clo6(3)  = zero
                clop6(3) = zero
              else

```

Reply:
For the first question, this is the old code:
```
          if(iclo6.eq.2) then
            xv(1,ia)=xv(1,ia)+clo6v(1,ia)
            yv(1,ia)=yv(1,ia)+clop6v(1,ia)
            xv(2,ia)=xv(2,ia)+clo6v(2,ia)
            yv(2,ia)=yv(2,ia)+clop6v(2,ia)
            sigmv(ia)=sigmv(ia)+clo6v(3,ia)
            dpsv(ia)=dpsv(ia)+clop6v(3,ia)
            xv(1,ia+1)=xv(1,ia+1)+clo6v(1,ia)
            yv(1,ia+1)=yv(1,ia+1)+clop6v(1,ia)
            xv(2,ia+1)=xv(2,ia+1)+clo6v(2,ia)
            yv(2,ia+1)=yv(2,ia+1)+clop6v(2,ia)
            sigmv(ia+1)=sigmv(ia+1)+clo6v(3,ia)
            dpsv(ia+1)=dpsv(ia+1)+clop6v(3,ia)
            oidpsv(ia)=one/(one+dpsv(ia))
            oidpsv(ia+1)=one/(one+dpsv(ia+1))
          else
            xv(1,ia)=xv(1,ia)+(clov(1,ia)*real(idz(1),fPrec))*          &
     &real(1-idfor,fPrec)    !hr05
            yv(1,ia)=yv(1,ia)+(clopv(1,ia)*real(idz(1),fPrec))*         &
     &real(1-idfor,fPrec)   !hr05
            xv(2,ia)=xv(2,ia)+(clov(2,ia)*real(idz(2),fPrec))*          &
     &real(1-idfor,fPrec)    !hr05
            yv(2,ia)=yv(2,ia)+(clopv(2,ia)*real(idz(2),fPrec))*         &
     &real(1-idfor,fPrec)   !hr05
            xv(1,ia+1)=xv(1,ia+1)+(clov(1,ia)*real(idz(1),fPrec))*      &
     &real(1-idfor,fPrec)  !hr05
            yv(1,ia+1)=yv(1,ia+1)+(clopv(1,ia)*real(idz(1),fPrec))*     &
     &real(1-idfor,fPrec) !hr05
            xv(2,ia+1)=xv(2,ia+1)+(clov(2,ia)*real(idz(2),fPrec))*      &
     &real(1-idfor,fPrec)  !hr05
            yv(2,ia+1)=yv(2,ia+1)+(clopv(2,ia)*real(idz(2),fPrec))*     &
     &real(1-idfor,fPrec) !hr05
          endif
          ejfv(ia)=e0f*(one+dpsv(ia))
          ejfv(ia+1)=e0f*(one+dpsv(ia+1))
          ejv(ia)=sqrt(ejfv(ia)**2+nucm0**2)                               !hr05
          ejv(ia+1)=sqrt(ejfv(ia+1)**2+nucm0**2)                           !hr05
          epsa(1)=(ampv(ia)**2/bet0v(ia,1))                              !hr05
          epsa(2)=(amp(2)**2/bet0v(ia,2))                                !hr05

          moidpsv(ia)=mtc(ia)/(one+dpsv(ia))
          moidpsv(ia+1)=mtc(ia+1)/(one+dpsv(ia+1))
          omoidpsv(ia)=c1e3*((one-mtc(ia))*oidpsv(ia))
          omoidpsv(ia+1)=c1e3*((one-mtc(ia+1))*oidpsv(ia+1))

```

Reply:
You are correct that the old code uses ia in both cases, but as far as I can tell the ia and ia+1 is the same value. Will double check this. My understanding is that the values only change for different machines (m), which is now always 1. I may have missed something though as this is all a bit convoluted.

Reply:
Discussed it with @tpersson, and this seems to be correct. Since these arrays are set from clo and clop, I can simplify the loop further, but it wouldn't change the results.

Reply:
ok let's leave the lines as they are - will come back on this later, if we will ever clean the code at such a level...